projects/graph/src/graph.py

```diff
@@ -6,4 +6,46 @@
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        self.vertices = {}
+    
+    def add_vertex(self, vertex):
+        self.vertices[vertex] = set()
+    
+    def add_edge(self, vert, edge2):
+        # specify which vertex
+        # receive inputs for edge cases regard that vertex
+        # update vertex set values
+        # print the dictionary
+        #code is below this line
+        # vert = f"Please specify vertex: {input()}"
+        
+        # self.vertices[vert].update(edge) 
+        # print(self.vertices)
+        if vert not in self.vertices:
+            raise Exception(f"No{vert} vertex")
+            self.vertices[vert].add(edge2)
+            if vert not in self.vertices:
+                raise Exception(f"No {edge2} vertex")
+            self.vertices[vert].add(edge2)
```

Comment:
You might have fixed this already, but there's no need to have three `self.vertices[vert].add(edge2)` here. Your `if..else..` statements can merely check for `vert` in `self.vertices`, and just raise errors as needed. Otherwise, execution will proceed to line 32 and establish the bidirectional edge as normal.